### PR TITLE
Feature/id 207/t 302

### PR DESCRIPTION
**Pull request by zrock-dev**
## Authors: 
- Gabriel Santiago Concha Saavedra

### **What?**
- Activity Transition after the user selects a role.

### **How?**
- Implementing logic in app to take a desition based on the user's role to which screen should the transition be made.

### **Why?**
- So we can change the behaviour of the application based on the role of the user.

### **Code reviewers/Testers**
@daniel-espinoza-79   
@VillalobosGarciaGaby  
@EmanuelJGalindo

### CODE REVIEW

File: app/src/main/java/com/isc/hermes/model/user/User.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.isc.hermes.model.user;
+
+public class User {
```
Comment (daniel-espinoza-79) [inline app/src/main/java/com/isc/hermes/model/user/User.java:3]: Missing JavaDocs

**Review by daniel-espinoza-79 — APPROVED** [review]
The presented code is well structured and complies with the clean code.

File: app/src/main/java/com/isc/hermes/MainActivity.java [hunk 2]
```diff
@@ -2,17 +2,21 @@
...
 import android.view.View;
 import android.widget.ImageButton;
+
 import com.isc.hermes.controller.SearcherController;
 import com.isc.hermes.controller.authentication.AuthenticationFactory;
 import com.isc.hermes.controller.authentication.AuthenticationServices;
 import com.isc.hermes.model.Searcher;
+
 import android.widget.LinearLayout;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/MainActivity.java:19]: Delete all the unnecessary space

File: app/src/main/java/com/isc/hermes/SignUpActivityView.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
 import com.isc.hermes.controller.authentication.AuthenticationFactory;
 import com.isc.hermes.controller.authentication.AuthenticationServices;
 import com.isc.hermes.controller.authentication.IAuthentication;
-import com.isc.hermes.model.User;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/SignUpActivityView.java:13]: Delete all the unnecessary space

File: app/src/main/java/com/isc/hermes/UserSignUpCompletionActivity.java [hunk 4]
```diff
@@ -30,6 +34,8 @@ public class UserSignUpCompletionActivity extends AppCompatActivity {
     private TextInputLayout comboBoxTextField;
     private Button buttonRegister;
     private ImageView imgUser;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/UserSignUpCompletionActivity.java:37]: Delete all the unnecessary space

File: app/src/main/java/com/isc/hermes/model/signup/RoleTransitionRepository.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    private static RoleTransitionRepository self;
+
+    private RoleTransitionRepository(){
+        rolesClassMap = new HashMap<>();
+
+        rolesClassMap.put(UserRoles.ADMINISTRATOR, EmailVerificationActivity.class);
+        rolesClassMap.put(UserRoles.GENERAL, MainActivity.class);
+    }
+
+    public static RoleTransitionRepository getInstance(){
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/model/signup/RoleTransitionRepository.java:26]: This method need javaDoc

File: app/src/main/java/com/isc/hermes/model/signup/SignUpTransitionHandler.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.isc.hermes.model.signup;
+
+import android.content.Context;
+import android.content.Intent;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/model/signup/SignUpTransitionHandler.java:5]: Delete all the unnecessary space

File: app/src/main/java/com/isc/hermes/model/user/User.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+    public String getPathImageUser(){
+        return payload.getPhotoURL();
+    }
+
+    public void setRole(UserRoles role){
+        this.role = role;
+    }
+
+    public UserRoles getRole() {
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/model/user/User.java:32]: this class and all method need javadoc

File: app/src/main/java/com/isc/hermes/model/user/UsersRepository.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    private static UsersRepository self;
+    private final HashMap<Integer, User> users;
+    private final TokenGenerator handler;
+
+    private UsersRepository(){
+        users = new HashMap<>();
+        handler = new TokenGenerator();
+    }
+
+    public static UsersRepository getInstance() {
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/model/user/UsersRepository.java:20]: this method need javadoc

File: app/src/test/java/com/isc/hermes/model/user/UserRolesTest.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package com.isc.hermes.model.user;
+
+import static org.junit.Assert.*;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/test/java/com/isc/hermes/model/user/UserRolesTest.java:4]: delete unnecessary space

**Review by VillalobosGarciaGaby — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/model/user/User.java (hunk 1 above)
Comment (zrock-dev) [reply app/src/main/java/com/isc/hermes/model/user/User.java:6]: Done

File: app/src/main/java/com/isc/hermes/MainActivity.java (hunk 2 above)
Comment (zrock-dev) [reply app/src/main/java/com/isc/hermes/MainActivity.java:19]: done

**Comment by zrock-dev** [comment]
@VillalobosGarciaGaby, most of the unnecessary spaces are being used as separators between group of imports that belong to the same library.

File: app/src/main/java/com/isc/hermes/model/user/User.java (hunk 7 above)
Comment (zrock-dev) [reply app/src/main/java/com/isc/hermes/model/user/User.java:63]: done

**Pull request merged**